velox/functions/lib/window/RowNumber.cpp

```diff
@@ -52,24 +61,35 @@ class RowNumberFunction : public exec::WindowFunction {
 
 } // namespace
 
-// Signature of this function is : row_number() -> bigint.
-void registerRowNumber(const std::string& name) {
+void registerRowNumber(
+    const std::string& name,
+    const TypeKind& offsetTypeKind) {
   std::vector<exec::FunctionSignaturePtr> signatures{
-      exec::FunctionSignatureBuilder().returnType("bigint").build(),
+      exec::FunctionSignatureBuilder()
+          .returnType(mapTypeKindToName(offsetTypeKind))
+          .build(),
   };
 
   exec::registerWindowFunction(
       name,
       std::move(signatures),
       [name](
           const std::vector<exec::WindowFunctionArg>& /*args*/,
-          const TypePtr& /*resultType*/,
+          const TypePtr& resultType,
           bool /*ignoreNulls*/,
           velox::memory::MemoryPool* /*pool*/,
           HashStringAllocator* /*stringAllocator*/,
           const core::QueryConfig& /*queryConfig*/)
           -> std::unique_ptr<exec::WindowFunction> {
-        return std::make_unique<RowNumberFunction>();
+        return std::make_unique<RowNumberFunction>(resultType);
       });
 }
-} // namespace facebook::velox::window::prestosql
+
+void registerIntegerRowNumber(const std::string& name) {
```

Comment:
Nit : Rename to registerRowNumberInteger and registerRowNumberBigint to be consistent with the rest of the naming.

Reply:
Done.